### PR TITLE
Fixed errors re Xcode 9.1/Swift4 in the charts-progress branch

### PR DESCRIPTION
**Pull request by bamse16**
Also fixed failing tests when the timezone is not GMT+1
And refactored send email to build the CSV data and body in a different method (that can be tested)
Fixed home screen background colour

### CODE REVIEW

File: BodyweightFitness/Domain/RepositoryRoutineCompanion.swift [hunk 1]
```diff
@@ -70,6 +70,104 @@ class RepositoryRoutineCompanion {
...
+                    }
+                }
+
+                index += 1
+            }
+        }
+
+        return content as String
+    }
+
```
Comment (mazurio) [inline BodyweightFitness/Domain/RepositoryRoutineCompanion.swift:167]: 👍 Brilliant

File: BodyweightFitness/Main/HomeViewController.swift [hunk 2]
```diff
@@ -8,6 +8,8 @@ class HomeViewController: AbstractViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
+
+        self.view.backgroundColor = UIColor.primary()
```
Comment (mazurio) [inline BodyweightFitness/Main/HomeViewController.swift:13]: This one now extends AbstractViewController - is that still a case?

File: BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift [hunk 3]
```diff
@@ -9,11 +9,41 @@ class RepositoryRoutineCompanionSpec: QuickSpec {
...
+        repositoryRoutine.exercises.append(secondExercise)
+        repositoryRoutine.exercises.append(thirdExercise)
+
+        repositoryRoutine.startTime = self.mockDate(from: "2017-08-07T13:13:00Z")
+        repositoryRoutine.lastUpdatedTime = self.mockDate(from: "2017-08-07T15:13:21Z")
+
+        repositoryRoutine.title = "Bodyweight Fitness"
+        repositoryRoutine.subtitle = "Recommended Routine"
+
+        return repositoryRoutine
```
Comment (mazurio) [inline BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift:44]: 👍

**Review by mazurio — APPROVED** [review]
Brilliant work. I haven't seen this PR before I merged/pushed my branch. Would you be able to rebase on top of master somehow? Happy to merge.

File: BodyweightFitness/Main/HomeViewController.swift (hunk 2 above)
Comment (bamse16) [reply BodyweightFitness/Main/HomeViewController.swift:13]: Yes, it belongs to a different view. I'll revert it if it's not the case.

**Comment by bamse16** [comment]
Let me figure out how to do that.

File: BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift [hunk 4]
```diff
@@ -213,6 +221,78 @@ class RepositoryRoutineCompanionSpec: QuickSpec {
...
+Bodyweight Fitness
+Recommended Routine
+
+
+Set 1, Seconds: 10
+
+
+Set 1, Seconds: 10
+
+
```
Comment (mazurio) [inline BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift:289]: Are exercise titles gone?

Comment (mazurio) [reply BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift:289]: Or maybe they are empty in the mock repository routine.

Comment (bamse16) [reply BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift:289]: They are empty in the sample text.

Comment (mazurio) [reply BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift:289]: That should be fine then, thanks for contributing! I am trying to make the code clearer and easier to contribute - still a lot to change to make it easier :-)

**Pull request closed without merging**